cpp/src/arrow/flight/client.h

```diff
@@ -41,6 +42,26 @@ namespace flight {
 
 class ClientAuthHandler;
 
+/// \brief A duration type for Flight call timeouts.
+typedef std::chrono::duration<double, std::chrono::seconds::period> TimeoutDuration;
+
+/// \brief Hints to the underlying RPC layer for Arrow Flight calls.
+class ARROW_EXPORT FlightCallOptions {
+ public:
+  /// Create a default set of call options.
+  FlightCallOptions();
+
+  /// \brief Set a timeout on this call. If the call does not complete
+  /// within the specified period, the call will complete with a
+  /// non-OK status.
+  void set_timeout(const TimeoutDuration& timeout_sec);
```

Comment:
Is this method still useful?

Reply:
Not really, it'll be more helpful to have some static factory methods instead. I'll make that change.